### DISCUSSION TITLE
Are there any resources based on our PhotoMaker? Please leave a link here and we will acknowledge and list it in the README.

### DISCUSSION THREAD

**Discussion by Paper99** [TencentARC/PhotoMaker#36] [Announcements]
Any form of resource is acceptable, including but not limited to Windows version, MacOS version, comfyui and WebUI versions. The only connection between them is PhotoMaker 😄

**Comment by yorickvP** [comment]
Hi! We've been working on https://replicate.com/jd7h/photomaker. We're currently working on some bugs and features and we'd like to transfer it be under the [TencentARC](https://replicate.com/tencentarc) org when it's done, but feel free to cherry-pick https://github.com/datakami-models/PhotoMaker/commit/db6f38fe7c8003069392979d210ada76d78f80c5 as a link in the meantime.

**Reply by Paper99** [reply]
Cool! We will update this in the README.

**Reply by Vkasdubey** [reply]
🔥🔥

**Comment by bmaltais** [comment]
I have updated some of your code to make the installation and running on Windows easier. Here is the current release in my fork. You might consider including some of the updates in your project to make it easy for Windows usage:

https://github.com/bmaltais/PhotoMaker/tree/v1.0.1

It also add support for non square output:

![image](https://github.com/TencentARC/PhotoMaker/assets/7474674/3c88b752-0776-4074-8a6e-fe6e36be4cb5)

I added a `.` at the end of the requirements-windows.txt to make it built the module during pip install.

**Reply by Paper99** [reply]
Cool! Thank you @bmaltais

**Reply by Paper99** [reply]
We'll update it.

**Reply by choigawoon** [reply]
cool this is what i want

**Reply by githubUser01946** [reply]
Why is it downloading things several times?

![image](https://github.com/TencentARC/PhotoMaker/assets/117453319/9beafba5-2348-41e6-895b-577089637a6c)

**Reply by mtangminghaiguge** [reply]
cool ！i am learnning...

**Comment by bmaltais** [comment]
@Paper99 Would it be possible to update the README.md link to my repo to:

```markdown
- [Windows version of PhotoMaker](https://github.com/bmaltais/PhotoMaker) by [@bmaltais](https://github.com/bmaltais), easy to deploy PhotoMaker on Windows. The description can be found in [this link](https://github.com/TencentARC/PhotoMaker/discussions/36#discussioncomment-8156199).
```

**Reply by Paper99** [reply]
Of course!

**Comment by bmaltais** [comment]
Here are the two ComfyUI node project I know about:

https://github.com/ZHO-ZHO-ZHO/ComfyUI-PhotoMaker

https://github.com/StartHua/Comfyui-Mine-PhotoMaker

**Reply by Paper99** [reply]
Thank you @bmaltais. We'll update them.

**Comment by ri131211** [comment]
Hello

**Comment by machineminded** [comment]
I'd like to integrate this into [Fooocus](https://github.com/lllyasviel/Fooocus), however their pipeline is customized.  Was hoping to get some guidance on this since Fooocus doesn't use `diffusers` out of the box.  Can the model be converted to a format expected by Fooocus?

Edit:  It has been integrated.  More features coming soon:

https://www.github.com/machineminded/Fooocus-inswapper

**Reply by Paper99** [reply]
Here is an implementation without `diffusers`:
https://github.com/shiimizu/ComfyUI-PhotoMaker.

**Reply by Paper99** [reply]
Merging to [Fooocus](https://github.com/lllyasviel/Fooocus) is really a cooool idea. 
Thank you for your efforts.

**Reply by choigawoon** [reply]
awesome!

**Reply by bmaltais** [reply]
You may want to watch out. I have done testing on that ComfyUI implementation and the results are not as good as the diffuser implementation. Specially on longer prompt where that likeness of face fade away…

probably related to a difference on how prompts and ksampler is applied?

But it is a good starting point…

**Reply by machineminded** [reply]
@Paper99 @bmaltais Very cool - thank you for your work.  I'm not too deep into the AI architecture but I should be able to integrate this into Fooocus.  Might take me some time so if someone else wants to take a stab at it feel free.

Originally I was going to just use it as `diffusers` in Fooocus, and it would just have its own special pipeline.  That still may be the best option given your experience @bmaltais

**Reply by Paper99** [reply]
Please let us know if you get the job done, thank you.

**Reply by machineminded** [reply]
I have a basic implementation for this here:  https://github.com/machineminded/Fooocus-inswapper/tree/feature/photomaker

I need to do some more testing, mostly to make sure pip package compatibility.  Otherwise I think the basic implementation is ready for use.  It has been implemented with `🤗 diffusers` and still hooks into the Fooocus async tasks.  Preview images don't look as nice as ksampler but I think it is a limitation of `diffusers`.

**Reply by Paper99** [reply]
Hi @machineminded,
Thanks for your efforts. 
We've updated your project in README file.
Commit message https://github.com/TencentARC/PhotoMaker/commit/78a0794e6218346c94bd8d3e8e45c0013e15e963

**Comment by woai18ytgjiang** [comment]
how to control two different face image mixing percentage in prompt?

**Comment by ToniDuque** [comment]
Does anyone know how to fix this after running pip install -r requirements-windows.txt?:
Building wheels for collected packages: photomaker
  Building wheel for photomaker (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for photomaker (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [28 lines of output]
      Traceback (most recent call last):
        File "C:\Users\Toni\Desktop\photomaker\venv\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Users\Toni\Desktop\photomaker\venv\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "C:\Users\Toni\Desktop\photomaker\venv\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "C:\Users\Toni\AppData\Local\Temp\pip-build-env-n7f5zrnf\overlay\Lib\site-packages\poetry\core\masonry\api.py", line 58, in build_wheel
          return WheelBuilder.make_in(
        File "C:\Users\Toni\AppData\Local\Temp\pip-build-env-n7f5zrnf\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 88, in make_in
          wb.build(target_dir=directory)
        File "C:\Users\Toni\AppData\Local\Temp\pip-build-env-n7f5zrnf\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 124, in build
          self._copy_module(zip_file)
        File "C:\Users\Toni\AppData\Local\Temp\pip-build-env-n7f5zrnf\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 269, in _copy_module
          to_add = self.find_files_to_add()
        File "C:\Users\Toni\AppData\Local\Temp\pip-build-env-n7f5zrnf\overlay\Lib\site-packages\poetry\core\masonry\builders\builder.py", line 166, in find_files_to_add
          for include in self._module.includes:
        File "C:\Users\Toni\AppData\Local\Programs\Python\Python310\lib\functools.py", line 981, in __get__
          val = self.func(instance)
        File "C:\Users\Toni\AppData\Local\Temp\pip-build-env-n7f5zrnf\overlay\Lib\site-packages\poetry\core\masonry\builders\builder.py", line 92, in _module
          return Module(
        File "C:\Users\Toni\AppData\Local\Temp\pip-build-env-n7f5zrnf\overlay\Lib\site-packages\poetry\core\masonry\utils\module.py", line 79, in __init__
          PackageInclude(
        File "C:\Users\Toni\AppData\Local\Temp\pip-build-env-n7f5zrnf\overlay\Lib\site-packages\poetry\core\masonry\utils\package_include.py", line 29, in __init__
          self.check_elements()
        File "C:\Users\Toni\AppData\Local\Temp\pip-build-env-n7f5zrnf\overlay\Lib\site-packages\poetry\core\masonry\utils\package_include.py", line 85, in check_elements
          raise ValueError(f"{root.name} is not a package.")
      ValueError: photomaker is not a package.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for photomaker
Failed to build photomaker
ERROR: Could not build wheels for photomaker, which is required to install pyproject.toml-based projects

**Comment by mhamzasani** [comment]
Is there an easy way to Install it and use it on Linux. I've tried hard on my server but run into errors. Want create a Rest API to use this? 

Need help!

**Reply by mhamzasani** [reply]
Also, does it only run on Large GPU? Can it run on Oracle Cloud 4 Ocpu?

**Reply by bmaltais** [reply]
If you can run ComfyUI then I suggest you install this custom node to make use of photomaker. Might work better with lower VRAM GPUs:  https://github.com/shiimizu/ComfyUI-PhotoMaker

It is the best node I could find out there as it integrate mostly using native ComfyUI nodes to load the LoRA part and create the image using the native ComfyUI KSampler.

**Comment by defisynner** [comment]
How can i input imgs of 2 diff people and have it in the prompt?

An example: uploading an image of a man and another image of a lady. And create a photo of them kissing

**Reply by Paper99** [reply]
Please refer to https://github.com/TencentARC/PhotoMaker/issues/112

**Reply by defisynner** [reply]
i tried having img twice in the prompt and it will reject my prompt

**Reply by Paper99** [reply]
Hello, you need to make modifications to pipeline.py. We currently cannot guarantee performance.

**Comment by bssrdf** [comment]
Hi, I have implemented PhotoMaker in [stable-diffusion.cpp](https://github.com/leejet/stable-diffusion.cpp/pull/179) using purely C/C++/CUDA. The results are not as good as the official version though, maybe because SD is not using the diffuser pipeline and I may have missed something. It does generate some interesting ID faces.

**Reply by Paper99** [reply]
Thanks for your awesome work.
We've added your work in [Related Resources](https://github.com/TencentARC/PhotoMaker?tab=readme-ov-file#purely-cccuda-version-of-photomaker).
